src/blueprint/azext_blueprint/tests/latest/test_blueprint_scenario.py

```diff
@@ -105,6 +116,29 @@ def test_blueprint(self, resource_group):
             '--name "{assignmentName}"',
             checks=[])
 
+        with mock.patch('azure.cli.command_modules.role.custom._gen_guid', side_effect=self.create_guid):
+            self.cmd(
+                'az role assignment create '
+                '--assignee-object-id {identityPrincipalId} '
+                '--role Owner '
+                '--scope /subscriptions/{subscription}/resourceGroups/{rgName}')
+
+        import time
+        time.sleep(600)

```

Comment:
magic number

Reply:
Yes, we may need better approach to wait for role assignment to take effective. I can't think of better ways at this moment.
https://docs.microsoft.com/en-us/azure/role-based-access-control/troubleshooting#role-assignment-changes-are-not-being-detected